core-bundle/src/Asset/ContaoContext.php

```diff
@@ -89,11 +99,32 @@ public function getStaticUrl(): string
 
     private function getPageModel(): ?PageModel
     {
-        if (isset($GLOBALS['objPage']) && $GLOBALS['objPage'] instanceof PageModel) {
+        $request = $this->requestStack->getCurrentRequest();
+
+        if (null === $request || !$request->attributes->has('pageModel')) {
+            return null;
+        }
+
+        $pageModel = $request->attributes->get('pageModel');
+
+        if ($pageModel instanceof PageModel) {
+            return $pageModel;
+        }
+
+        if (
```

Comment:
This if condition will never be evaluated if `$request` is `null` or does not have an attribute named `pageModel`. Intended?

Reply:
yes, the if-condition refers to `(int) $pageModel` (from the request), we can't operate without that. The method is not supposed to _always_ get the (global) page model, but to return the one from the current request.